frontend/src/pages/PipelineDetails.tsx

```diff
@@ -600,12 +603,13 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
   public async handleVersionSelectedV2(versionId: string): Promise<void> {
```

Comment:
nit: Out of scope for the bug fix, but the two handlers looks highly likely, maybe we should consider some refactoring work to reduce the code duplication?

Reply:
Great idea! Will do.

Reply:
Refactor in https://github.com/kubeflow/pipelines/pull/9694